services/streammanager/wunderkind/wunderkindmanager.go

```diff
@@ -0,0 +1,111 @@
+package wunderkind
+
+import (
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/cmd/rudder-cli/config"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+const (
+	INVOCATION_TYPE         = "RequestResponse"
+	WUNDERKIND_REGION       = "WUNDERKIND_REGION"
+	WUNDERKIND_IAM_ROLE_ARN = "WUNDERKIND_IAM_ROLE_ARN"
+	WUNDERKIND_EXTERNAL_ID  = "WUNDERKIND_EXTERNAL_ID"
+	WUNDERKIND_LAMBDA       = "WUNDERKIND_LAMBDA"
+)
+
+var (
+	pkgLogger logger.Logger
+	jsonfast  = jsoniter.ConfigCompatibleWithStandardLibrary
+)
+
+type inputData struct {
+	Payload string `json:"payload"`
+}
+
+type WunderkindProducer struct {
+	client LambdaClient
+}
+
+type LambdaClient interface {
+	Invoke(input *lambda.InvokeInput) (*lambda.InvokeOutput, error)
+}
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("streammanager").Child(lambda.ServiceName)
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*WunderkindProducer, error) {
+	sessionConfig := &awsutil.SessionConfig{
+		Region:        config.GetEnv(WUNDERKIND_REGION),
+		IAMRoleARN:    config.GetEnv(WUNDERKIND_IAM_ROLE_ARN),
+		ExternalID:    config.GetEnv(WUNDERKIND_EXTERNAL_ID),
+		RoleBasedAuth: true,
+	}
+	awsSession, err := awsutil.CreateSession(sessionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &WunderkindProducer{client: lambda.New(awsSession)}, nil
+}
+
+// Produce creates a producer and send data to Lambda.
+func (producer *WunderkindProducer) Produce(jsonData json.RawMessage, destConfig interface{}) (int, string, string) {
+	client := producer.client
+	if client == nil {
+		return 400, "Failure", "[Wunderkind] error :: Could not create client"
+	}
+
+	var input inputData
+	err := jsonfast.Unmarshal(jsonData, &input)
+	if err != nil {
+		returnMessage := "[Wunderkind] error while unmarshalling jsonData :: " + err.Error()
+		return 400, "Failure", returnMessage
+	}
+	if input.Payload == "" {
+		return 400, "Failure", "[Wunderkind] error :: Invalid payload"
+	}
+
+	var invokeInput lambda.InvokeInput
+	wunderKindLambda := config.GetEnv(WUNDERKIND_LAMBDA)
```

Comment:
DI config

Reply:
Can you provide some more clarification on this?

Reply:
You would need to take config as a parameter in the constructor and store it as part of Producer struct. Later you can us it `producer.conf.GetString(WunderkindLambda, "")`
```GO
// NewProducer creates a producer based on destination config
func NewProducer(conf *config.Config) (*Producer, error) {
	awsSession, err := awsutil.CreateSession(&awsutil.SessionConfig{
		Region:        conf.GetString(WunderkindRegion, ""),
		IAMRoleARN:    conf.GetString(WunderkindIamRoleArn, ""),
		ExternalID:    conf.GetString(WunderkindExternalId, ""),
		RoleBasedAuth: true,
	})
	if err != nil {
		return nil, fmt.Errorf("creating session: %w", err)
	}

	return &Producer{
		conf:   conf,
		logger: logger.NewLogger().Child("streammanager").Child("wunderkind"),
		client: lambda.New(awsSession),
	}, nil
}
```

Reply:
addressed.
`I have used config.default in caller function of NewProducer`